### PR TITLE
Use golangci lint for linting

### PR DESCRIPTION
**Pull request by asalkeld**

### CODE REVIEW

File: controllers/clusteroperator.go [hunk 1]
```diff
@@ -114,15 +113,6 @@ func setStatusCondition(conditionType osconfigv1.ClusterStatusConditionType,
...
 
-// getOperandVersions returns current version
-func getOperandVersions() []osconfigv1.OperandVersion {
-	operandVersions := []osconfigv1.OperandVersion{}
-	if releaseVersion := os.Getenv("RELEASE_VERSION"); len(releaseVersion) > 0 {
-		operandVersions = append(operandVersions, osconfigv1.OperandVersion{Name: "operator", Version: releaseVersion})
-	}
-	return operandVersions
-}
-
```
Comment (asalkeld) [inline controllers/clusteroperator.go:125]: unused function

File: controllers/clusteroperator_test.go [hunk 2]
```diff
@@ -5,15 +5,15 @@ import (
 	"testing"
 	"time"
 
+	osconfigv1 "github.com/openshift/api/config/v1"
```
Comment (kirankt) [inline controllers/clusteroperator_test.go:8]: Why was this moved? Usually we're keeping the imports in the following format:

std libs

outside libs

openshift libs

File: controllers/clusteroperator.go (hunk 1 above)
Comment (kirankt) [reply controllers/clusteroperator.go:125]: Is the linter complaining about this?

File: controllers/clusteroperator_test.go [hunk 3]
```diff
@@ -45,7 +45,10 @@ func TestUpdateCOStatusDisabled(t *testing.T) {
 	reconciler.OSClient = fakeconfigclientset.NewSimpleClientset(co)
 
 	for _, tc := range tCases {
-		reconciler.updateCOStatusDisabled()
+		err := reconciler.updateCOStatusDisabled()
```
Comment (kirankt) [inline controllers/clusteroperator_test.go:48]: Is this linting related? If not, we should not worry about it in this PR.

File: controllers/provisioning_controller.go [hunk 4]
```diff
@@ -20,15 +20,15 @@ import (
 	"fmt"
 
 	"github.com/go-logr/logr"
+	osconfigv1 "github.com/openshift/api/config/v1"
```
Comment (kirankt) [inline controllers/provisioning_controller.go:23]: Same here. Need to move the openshift libs down.

File: controllers/provisioning_controller_test.go [hunk 5]
```diff
@@ -3,24 +3,24 @@ package controllers
 import (
 	"testing"
 
+	configv1 "github.com/openshift/api/config/v1"
```
Comment (kirankt) [inline controllers/provisioning_controller_test.go:6]: Ditto.

File: main.go [hunk 6]
```diff
@@ -19,6 +19,8 @@ import (
 	"flag"
 	"os"
 
+	osconfigv1 "github.com/openshift/api/config/v1"
```
Comment (kirankt) [inline main.go:22]: Another one.

**Review by kirankt — CHANGES_REQUESTED** [review]
Few minor nits and also a question about changes not specifically related to this linting change. Please let me know what you think.

File: controllers/clusteroperator.go (hunk 1 above)
Comment (asalkeld) [reply controllers/clusteroperator.go:125]: yes, don't worry tho' when we need it, it will be in git history..

File: controllers/clusteroperator_test.go (hunk 3 above)
Comment (asalkeld) [reply controllers/clusteroperator_test.go:48]: yes, it forces you to check errors

File: controllers/clusteroperator_test.go (hunk 2 above)
Comment (asalkeld) [reply controllers/clusteroperator_test.go:8]: aaah, OK so goimports does
```
std libs

outside libs

local libs
```
I can adjust the definition of "local" to github.com/openshift

Comment (asalkeld) [reply controllers/clusteroperator_test.go:8]: aaah, OK so goimports does
```
std libs

outside libs

local libs
```
I can adjust the definition of "local" to github.com/openshift

**Review by kirankt — APPROVED** [review]
Thanks for addressing changes. Looks fine to me. I'll let either @dhellmann, @sadasu or @andfasano  take a look and lgtm this PR.
